src/Spectre.Database/Utils/DatasetDetailsFinder.cs

```diff
@@ -0,0 +1,119 @@
+﻿/*
+ * DatasetDetailsFinder.cs
+ * Class for translating hash and friendly name to name of the file.
+ *
+   Copyright 2017 Roman Lisak
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+namespace Spectre.Database.Utils
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data.Entity;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Spectre.Database.Contexts;
+    using Spectre.Database.Entities;
+
+    /// <summary>
+    /// Class for finding the path of chosen property.
+    /// </summary>
+    public class DatasetDetailsFinder : IDatasetDetailsFinder
+    {
+        /// <summary>
+        /// The context used for queries.
+        /// </summary>
+        private DatasetsContext _context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DatasetDetailsFinder"/> class.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        public DatasetDetailsFinder(DatasetsContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// Query for translating hash to upload number.
+        /// </summary>
+        /// <param name="hash">The hash.</param>
+        /// <returns>
+        /// Returns UploadNumber for Hash.
```

Comment:
Documentation should specify what is returned when hash is not found. It goes same for all the methods below.

Reply:
We should also go for .NET standard-conforming names in such a case, like `HashToUploadNumberOrDefault`, besides of entry in the docstrings, as stated in the conversation with @KapRom.